src/datachain/sql/types.py

```diff
@@ -328,7 +396,11 @@ def uint64(self, value):
         return value
 
     def float(self, value):
-        return float("nan") if value is None else value
+        if value is None:
+            return float("nan")
+        if isinstance(value, str) and value.lower() == "nan":
+            return float("nan")
```

Comment:
Were you encountering this with either sqlite or clickhouse?

Reply:
It is SQLite.

Reply:
SQLite CLI:

```
$ sqlite3 .datachain/db
SQLite version 3.43.2 2023-10-10 13:08:14
Enter ".help" for usage hints.
sqlite> create table foo (id int, val float not null default nan);
sqlite> insert into foo (id) values (1);
sqlite> select * from foo;
1|nan
sqlite> 
$
```

Python:

```
$ ipython
Python 3.12.5 (main, Aug  6 2024, 19:08:49) [Clang 15.0.0 (clang-1500.3.9.4)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.25.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sqlite3

In [2]: conn = sqlite3.connect('.datachain/db')

In [3]: cur = conn.cursor()

In [4]: cur.execute('select * from foo')
Out[4]: <sqlite3.Cursor at 0x11064f4c0>

In [5]: cur.fetchall()
Out[5]: [(1, 'nan')]

In [6]:
Do you really want to exit ([y]/n)? ^D
$
```

Reply:
Not sure there is a better solution, but want to note some edge cases when setting the default to `nan` in sqlite:
- unexpected sorting and equality operations as discussed in https://stackoverflow.com/questions/20619957/nan-in-sqlite3-have-invalid-numerical-comparisons
- sqlite stores nan values like 10/0 as nulls in sqlite rather than `nan`, which again may cause unexpected results like null and nan being sorted differently

Reply:
Absolutely agree 👍 Hopefully we will not need default values in SQLite.